docs/preview/features/openapi/security-definitions.md

```diff
@@ -18,21 +22,61 @@ PM > Install-Package Arcus.WebApi.OpenApi.Extensions
 
 ## Usage
 
-To indicate that an API is protected by OAuth, you need to add `AuthorizeCheckOperationFilter` as an `OperationFilter` when configuring Swashbuckles Swagger generation:
+### OAuth
+
+To indicate that an API is protected by OAuth, you need to add `OAuthAuthorizeOperationFilter` as an `IOperationFilter` when configuring Swashbuckles Swagger generation:
+
+```csharp
+services.AddSwaggerGen(setupAction =>
+{
+    setupAction.SwaggerDoc("v1", new Info { Title = "My API v1", Version = "v1" });
+
+    setupAction.AddSecurityDefinition("oauth2", new OAuth2Scheme
+    {
+        Flow = "implicit",
+        AuthorizationUrl = $"{authorityUrl}connect/authorize",
+        Scopes = scopes
+    });
+    
+    setupAction.OperationFilter<OAuthAuthorizeOperationFilter>(new object[] { new[] { "myApiScope1", "myApiScope2" } });
+});
+```
+
+### Shared Access Key
+
+To indicate that an API is protected by [Shared Access Key authentication](../../features/security/auth/shared-access-key), you need to add `SharedAccessKeyAuthenticationOperationFilter` as an `IOperationFilter` when configuring Swashbuckles Swagger generation:
+
+```csharp
+services.AddSwaggerGen(setupAction =>
+{
+    setupAction.SwaggerDoc("v1", new Info { Title = "My API v1", Version = "v1" });
+
+    setupAction.AddSecurityDefinition("sharedaccesskey", new OpenApiSecurityScheme
+    {
+        Name = "X-API-Key",
+        Type = SecuritySchemeType.ApiKey,
+        In = ParameterLocation.Header
+     });
+
+     setupAction.OperationFilter<SharedAccessKeyAuthenticationOperationFilter>();
+});
+```
+
+### Certificate
+
+To indicate that an API is protected by [Certificate authentication](../../features/security/auth/certificate), you need to add `CertificateAuthenticationOperationFilter` as an `IOperationFilter` when configuring Swashbuckles Swagger generation:
 
 ```csharp
 services.AddSwaggerGen(setupAction =>
 {
-   setupAction.SwaggerDoc("v1", new Info { Title = "My API v1", Version = "v1" });
+    setupAction.SwaggerDoc("v1", new Info { Title = "My API v1", Version = "v1" });
 
-   setupAction.AddSecurityDefinition("oauth2", new OAuth2Scheme
-   {
-      Flow = "implicit",
-      AuthorizationUrl = $"{authorityUrl}connect/authorize",
-      Scopes = scopes
-   });
+    setupAction.AddSecurityDefinition("certificate", new OpenApiSecurityScheme
+    {
+        Type = SecuritySchemeType.ApiKey
+     });
 
-   setupAction.OperationFilter<OAuthAuthorizeOperationFilter>(new object[] {new [] {"myApiScope1", "myApiScope2"});
+     setupAction.OperationFilter<CertificateAuthenticationOperationFilter>();
```

Comment:
Same here, are we assuming `certificate` or is it configurable? What happens if I use a different name?

Reply:
Made it configurable.